src/main/java/edu/kit/compiler/semantic/SemanticChecks.java

```diff
@@ -184,7 +184,8 @@ public Boolean visit(ArrayAccessExpressionNode expr) {
     }
 
     public Boolean visit(IdentifierExpressionNode expr) {
-        if (isMain && expr.getDefinition().getKind() == DefinitionKind.Parameter) {
+        if (isMain && !expr.isHasError() && expr.getDefinition()
+                .getKind() == DefinitionKind.Parameter) {
```

Comment:
Correct, but that's an interesting formatting

Reply:
:disappointed: